java/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ClientWorker.java

```diff
@@ -68,12 +70,10 @@ public ClientWorker(TargetConfiguration targetConfiguration,
         this.targetConfiguration = targetConfiguration;
         this.response = response;
         this.expectEntityBody = response.isExpectResponseBody();
-
-        Map<String,String> headers = response.getHeaders();
+        @SuppressWarnings("startswith")  Map<String, @StartsWith({"https", "file"})String> headers = response.getHeaders();
```

Comment:
I don't think this annotation is correct - all of the headers needn't start with `"https"`. It seems like there's some code below (which you've marked as a true positive) which is at least intended to convert http to https.

Reply:
Would like to talk to you about this so left this one out.

Reply:
Headers I think should all start with https as can be seen inhttps://docs.oracle.com/javase/7/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/Headers.html. They are either an HTTP request or response so they should start with https but they most likely won't so should be a true positive according to me

Reply:
Can you change this to reference Java 8 documentation, since this is Java 8 code? Otherwise this seems fine.